tests/python/test_base.py

```diff
@@ -21,6 +24,22 @@ def setUp(self):
             os.path.join(os.path.dirname(__file__), "..", "fixtures")
         )
 
+    def mock_jira_session_bases(self):
+        """
+        Replace the JiraSession base class to be MockedJira
+        so we can run tests without an actual connection to
+        JIRA.
+        """
+        # Do not use mock.patcher for this, as the patcher's exit
+        # method will try to delete the bases before restoring them,
+        # which will raise an error.
+        self.__old_bases = JiraSession.__bases__
```

Comment:
Some tests need to use the JIRASession class, which derives from the JIRA class, but since the tests happen offline, we're changing the base class to be the mocked version of JIRA. The original test code tried to use mock.patch, but that caused an error when unpacthing due to how the mock module unpaches values. Therefore I've written a poor's man version of monkey patching for that class.